test/e2e/network_segmentation_endpointslices_mirror.go

```diff
@@ -44,186 +44,204 @@ var _ = Describe("Network Segmentation EndpointSlices mirroring", func() {
-					esList, err := cs.DiscoveryV1().EndpointSlices(f.Namespace.Name).List(context.TODO(), metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", "k8s.ovn.org/service-name", svc.Name)})
-					if err != nil {
-						return err
-					}
-
-					if len(esList.Items) != 0 {
-						return fmt.Errorf("expected no mirrored EndpointSlice, got: %d", len(esList.Items))
-					}
-					return nil
-				}, 2*time.Minute, 6*time.Second).ShouldNot(HaveOccurred())
-
 			},
-			Entry(
-				"L2 dualstack primary UDN, cluster-networked pods",
-				networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer2",
-					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				},
-				false,
-			),
-			Entry(
-				"L3 dualstack primary UDN, cluster-networked pods",
-				networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer3",
-					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				},
-				false,
-			),
-			Entry(
-				"L2 dualstack primary UDN, host-networked pods",
-				networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer2",
-					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				},
-				true,
-			),
-			Entry(
-				"L3 dualstack primary UDN, host-networked pods",
-				networkAttachmentConfigParams{
-					name:     nadName,
-					topology: "layer3",
-					cidr:     fmt.Sprintf("%s,%s", userDefinedNetworkIPv4Subnet, userDefinedNetworkIPv6Subnet),
-					role:     "primary",
-				},
-				true,
-			),
+			Entry("NetworkAttachmentDefinitions", func(c networkAttachmentConfigParams) error {
+				netConfig := newNetworkAttachmentConfig(c)
+				nad := generateNAD(netConfig)
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(context.Background(), nad, metav1.CreateOptions{})
+				return err
+			}),
+			Entry("UserDefinedNetwork", func(c networkAttachmentConfigParams) error {
+				udnManifest := generateUserDefinedNetworkManifest(&c)
+				cleanup, err := createManifest(f.Namespace.Name, udnManifest)
+				DeferCleanup(cleanup)
+				Expect(waitForUserDefinedNetworkReady(f.Namespace.Name, c.name, 5*time.Second)).To(Succeed())
+				return err
+			}),
 		)
 
-		DescribeTable(
-			"does not mirror EndpointSlices in namespaces not using user defined primary networks",
-			func(
-				netConfigParams networkAttachmentConfigParams,
-			) {
-				netConfig := newNetworkAttachmentConfig(netConfigParams)
-
-				netConfig.namespace = f.Namespace.Name
-
-				By("creating the attachment configuration")
-				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
-					context.Background(),
-					generateNAD(netConfig),
-					metav1.CreateOptions{},
+		DescribeTableSubtree("created using",
+			func(createNetworkFn func(c networkAttachmentConfigParams) error) {
```

Comment:
It seems `createNetworkFn` is not called in this tests

Reply:
good catch, copy-paste error
https://github.com/ovn-org/ovn-kubernetes/pull/4643/commits/4a2d447a68233d152d7b6971fc0b93363b69bdb4#diff-efa6082185cdcbcd4b0ef97df7abbe8ac5013a23b8530d5714e47845c4a272a1R179
Fixed.